cloudigrade/analyzer/management/commands/analyze_log.py

```diff
@@ -136,3 +170,36 @@ def _is_on_off_event(self, record, valid_events):
             return False
         else:
             return True
+
+    @transaction.atomic
+    def _save_instances_and_events(self, instances):
+        """
+        Save instances and on/off events to the DB.
+
+        Args:
+            instances (dict): Instance data, including a list of events.
+
+        """
+        for instance_id, instance_dict in instances.items():
+            account_id = instance_dict['account_id']
+            # TODO: Don't pull the same account record from the
+            #       DB multiple times. Consider caching.
+            account = Account.objects.get(account_id=account_id)
+            instance, __ = Instance.objects.get_or_create(
+                account=account,
+                ec2_instance_id=instance_id,
+                region=instance_dict['region'],
+            )
+            for event in instance_dict['events']:
+                occurred_at_dt = datetime.datetime.strptime(
+                    event['occurred_at'], '%Y-%m-%dT%H:%M:%SZ'
```

Comment:
Just me not knowing AWS's internals here…

Does AWS always use the `Z` time zone in its timestamps?

Reply:
Fairly certain, yes. Is there a safer way to approach this?

Reply:
An alternative would be to use `dateutil` because it tries to be _clever_ about parsing date strings, but it we're fairly certain and don't expect ambiguity, then let's stick with what you have! 👍 

For completion sake, here's an example of what `dateutil`'s behavior looks like:
```
>>> from dateutil import parser
>>> parser.parse('2000-01-01T01:23:45')
datetime.datetime(2000, 1, 1, 1, 23, 45)
>>> parser.parse('2000-01-01T01:23:45Z')
datetime.datetime(2000, 1, 1, 1, 23, 45, tzinfo=tzutc())
>>> parser.parse('2000-01-01T01:23:45GMT')
datetime.datetime(2000, 1, 1, 1, 23, 45, tzinfo=tzutc())
>>> parser.parse('2000-01-01T01:23:45UTC')
datetime.datetime(2000, 1, 1, 1, 23, 45, tzinfo=tzutc())
>>> parser.parse('2000-01-01T01:23:45+0500')
datetime.datetime(2000, 1, 1, 1, 23, 45, tzinfo=tzoffset(None, 18000))
>>> parser.parse('2000-01-01T01:23:45-5')
datetime.datetime(2000, 1, 1, 1, 23, 45, tzinfo=tzoffset(None, -18000))
```

Reply:
I haven't used the parser before, so thanks for shedding 💡 on it. Putting in my toolbox. I do ❤️ `tz` and `relativedelta`.